be/src/exec/pipeline/scan/olap_schema_chunk_source.cpp

```diff
@@ -0,0 +1,98 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#include "exec/pipeline/scan/olap_schema_chunk_source.h"
+
+#include <boost/algorithm/string.hpp>
+
+#include "exec/vectorized/schema_scanner.h"
+#include "exec/workgroup/work_group.h"
+
+namespace starrocks {
+namespace pipeline {
+
+OlapSchemaChunkSource::OlapSchemaChunkSource(int32_t scan_operator_id, RuntimeProfile* runtime_profile,
+                                             MorselPtr&& morsel, OlapSchemaScanContextPtr ctx)
+        : ChunkSource(scan_operator_id, runtime_profile, std::move(morsel), ctx->get_chunk_buffer()), _ctx(ctx) {}
+
+OlapSchemaChunkSource::~OlapSchemaChunkSource() {}
+
+Status OlapSchemaChunkSource::prepare(RuntimeState* state) {
+    RETURN_IF_ERROR(ChunkSource::prepare(state));
+    _dest_tuple_desc = state->desc_tbl().get_tuple_descriptor(_ctx->tuple_id());
+    if (_dest_tuple_desc == nullptr) {
+        return Status::InternalError("failed to get tuple descriptor");
+    }
+    const SchemaTableDescriptor* schema_table =
+            static_cast<const SchemaTableDescriptor*>(_dest_tuple_desc->table_desc());
+    if (schema_table == nullptr) {
+        return Status::InternalError("Failed to get schema table descriptor");
+    }
+
+    auto param = _ctx->param();
+    param->_rpc_timer = ADD_TIMER(_runtime_profile, "FERPC");
+    param->_fill_chunk_timer = ADD_TIMER(_runtime_profile, "FillChunk");
+
+    _filter_timer = ADD_TIMER(_runtime_profile, "FilterTime");
+
+    _scanner = vectorized::SchemaScanner::create(schema_table->schema_table_type());
+    if (_scanner == nullptr) {
+        return Status::InternalError("schema scanner get null pointer");
+    }
+
+    RETURN_IF_ERROR(_scanner->init(param, _ctx->object_pool()));
+    return _scanner->start(state);
+}
+
+void OlapSchemaChunkSource::close(RuntimeState* state) {}
```

Comment:
don't need we need to do anything when close?

Reply:
> don't need we need to do anything when close?

Yes, we don't need to do anything actively, using the destructor to release resources is enough.

Reply:
when the instance is reused, but we should reset its status, so close() shoulc clear some context.

Reply:
> when the instance is reused, but we should reset its status, so close() shoulc clear some context.

It seems that ChunkSource won't be reused, in scan operator, we will create a new ChunkSource for each morsel.